### PR TITLE
hook scaffolding

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes: #14341,#8366,#8379,#8378,#8375,#8373,#8371,#8368

This PR scaffolds the (hopefully) remaining hooks to migrate to react-query. 

- Filtering logic was moved to a dedicated function in utils/filters. 
- `authorizedFetch` was folded into the Session Context and now allows manual urls in addition to the endpointconfig object.
- The organization settings list page uses the new hook. 
- Cleanup of folder structure for api hooks
- Cleanup of mocks for hooks
- Import path updates from cleanup

### CODE REVIEW

File: frontend-react/src/hooks/api/organizations/UseDeleteOrganizationReceiver/UseDeleteOrganizationReceiver.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    const fn = (
+        obj: Partial<Omit<RSReceiver, "name" | "organizationName">> & {
+            name: string;
+            organizationName: string;
+        },
+    ) => {
+        return authorizedFetch({
+            url: `/settings/organizations/${obj.organizationName}/receivers/${obj.name}`,
+            method: "delete",
```
Comment (penny-lischer) [inline frontend-react/src/hooks/api/organizations/UseDeleteOrganizationReceiver/UseDeleteOrganizationReceiver.ts:17]: Can we use the enum here? HTTPMethods.DELETE

File: frontend-react/src/contexts/Session/SessionProvider.tsx [hunk 2]
```diff
@@ -58,6 +65,65 @@ export interface SessionProviderProps extends PropsWithChildren {
...
+
+export async function staticAuthorizedFetch<T = unknown>({
+    apiUrl,
+    sessionId = "",
+    accessToken = "",
+    organization = "",
+    options,
+    endpointConfig,
+}: StaticAuthorizedFetchProps) {
+    if (options.segments && !endpointConfig)
```
Comment (etanb) [inline frontend-react/src/contexts/Session/SessionProvider.tsx:85]: Do we want to log any of these potential errors to Azure?

File: frontend-react/src/hooks/api/UseActionHistory/UseActionHistory.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { useSuspenseQuery } from "@tanstack/react-query";
```
Comment (etanb) [inline frontend-react/src/hooks/api/UseActionHistory/UseActionHistory.ts:1]: Unless I'm mistaken, `useActionHistory` isn't used anywhere? This is setting up for a future PR? If so, can we add some comments?

File: frontend-react/src/contexts/Session/SessionProvider.tsx [hunk 4]
```diff
@@ -5,6 +5,7 @@ import {
     UserClaims,
 } from "@okta/okta-auth-js";
 import { useOktaAuth } from "@okta/okta-react";
+import axios, { AxiosError } from "axios";
```
Comment (etanb) [inline frontend-react/src/contexts/Session/SessionProvider.tsx:8]: I understand this _might_ be out of the scope of this PR, but we do use both `fetch` and `axios` in our project. Is it time that we hard cut to one or the other? 

IE in: AdminOrgEdit.tsx AdminLastMileFailuresTable.tsx EditSenderSettings.tsx

Comment (jpandersen87) [reply frontend-react/src/contexts/Session/SessionProvider.tsx:8]: Yes. Axios is currently the pattern (for context, established even before me), which is why it is continuing to be used. Getting everything in the project moved to using these new hooks will naturally align the project in one direction.

File: frontend-react/src/hooks/api/UseActionHistory/UseActionHistory.ts (hunk 3 above)
Comment (jpandersen87) [reply frontend-react/src/hooks/api/UseActionHistory/UseActionHistory.ts:1]: Aside from the existing hooks that were reorganized, all of these are unused. This is scaffolding by getting the bulk of remaining ones into the project to quickly iterate on more nuanced PRs to use them in pages.

**Review by penny-lischer — APPROVED** [review]
lgtm

File: frontend-react/src/hooks/api/UseActionHistory/UseActionHistory.ts (hunk 3 above)
Comment (etanb) [reply frontend-react/src/hooks/api/UseActionHistory/UseActionHistory.ts:1]: Still might be good to at least add a comment on each denoting that?

**Review by etanb — APPROVED** [review]
Small comment about commenting, but 👍

**Pull request merged**